src/Connection.js

```diff
@@ -18,7 +18,9 @@ export default class Connection {
 
         if (handle[key].is_pool) {
             // handle._logger.log('Added a pool connection for', key);
-            connection = mysql.createPool(handle[key].config);
+            connection = !handle[key].connection
+                       ? mysql.createPool(handle[key].config)
+                       : handle[key].connection;
```

Comment:
Why not just 

connection = handle[key].connection || mysql.createPool(handle[key].config);

Also, the way storing Connection object and Pool object in the same (handler.connection) is confusing to begin with. Should have stored it in a separate field, and make the connection just a mysql connection.
https://github.com/anyTV/anytv-node-mysql/blob/master/src/Transaction.js#L53